File: lecture6/pset6/cash.py
import cs50

cents = 0
coins = 0
qrt = 25
dm = 10
nc = 5
pn = 1

while True:
    change = cs50.get_float("Change owed: ") * 100
    
    if change >= 1:
        break
    
while cents + qrt <= change:
    coins = coins + 1
    cents = cents + qrt
    
while cents + dm <= change:
    coins = coins + 1
    cents = cents + dm
    
while cents + nc <= change:
    coins = coins + 1
    cents = cents + nc
    
while cents + pn <= change:
    coins = coins + 1
    cents = cents + pn
    
print(coins)

File: lecture1/pset1/cash1/cash.c
#include <cs50.h> 
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    int coins = 0;
    int cents = 0;
    int qrt = 25;
    int dm = 10; 
    int nckl = 5; 
    int pnn = 1;
    
    // Asking for the change owed:
    do 
    {
        change = round(get_float("Change owed: ") * 100);
    } while (change < 1);
    
    // Calculating how many coins it needs:
    while (cents + qrt <= change) 
    {
        coins++;
        cents += qrt; 
    } 
    
    while (cents + dm <= change) 
    {
        coins++;
        cents += dm;
    } 
    
    while (cents + nckl <= change)
    {
        coins++;
        cents += nckl;
    }
    
    while (cents + pnn <= change)
    {
        coins++;
        cents += pnn;
    }
    
    // Just printing how many coins:
    printf("%i\n", coins);
}